actionpack/lib/action_controller/test_case.rb

```diff
@@ -208,10 +207,14 @@ def load!
   end
 
   # Superclass for ActionController functional tests. Functional tests allow you to
-  # test a single controller action per test method. This should not be confused with
-  # integration tests (see ActionDispatch::IntegrationTest), which are more like
-  # "stories" that can involve multiple controllers and multiple actions (i.e. multiple
-  # different HTTP requests).
+  # test a single controller action per test method.
+  #
+  # == Use `ActionDispatch::Integration` tests instead of this.
+  #
+  # Rails discourages use of `ActionController` functional tests in favor of integration tests
+  # (see ActionDispatch::IntegrationTest).
+  # New Rails applications no longer generate `ActionController` style functional tests and they should
```

Comment:
I still disagree that we're referencing the constant here. In my opinion we are talking about Action Controller the library.